### PR TITLE
Encoding support

### PR DESCRIPTION
**Pull request by tonila**
Hi,

Came across this beayty, but I needed encoding. So here is my pull request.

### CODE REVIEW

**Comment by tonila** [comment]
Reimplemented my first implementation with options API.

`constructor(file: File, options: Options={})`

That triggered multiple pushes, since I haven't done much with Typescipt, but I think I got is somewhat right now.

File: src/index.ts [hunk 1]
```diff
@@ -12,14 +13,17 @@ class LineReader {
 	private readPosition: number; // Position of read head
 	private chunk: string; // Current chunk text contents
 	private lines: string[]; // Array of current lines read
+	private options: Options; // Options
 
-	public constructor(file: File) {
+	public constructor(file: File, options: Options={}) {
```
Comment (stanrogo) [inline src/index.ts]:
```suggestion
	public constructor(file: File, options: Options = { encoding: 'UTF-8' }) {
```

File: src/index.ts [hunk 2]
```diff
@@ -12,14 +13,17 @@ class LineReader {
...
 		this.fileReader = new FileReader();
 		this.readPosition = 0;
 		this.chunk = '';
 		this.lines = [];
 		this.file = file;
 		this.events = new Map<string, Function>();
+		if (!options.encoding) options.encoding = 'UTF-8'
```
Comment (stanrogo) [inline src/index.ts]: You can then remove this line since it's already in the argument defaults

File: src/index.ts (hunk 1 above)
Comment (stanrogo) [reply src/index.ts]: This is neater in my opinion

File: src/index.d.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 // Type definitions for browser-line-reader 0.02
 // Project: browser-line-reader
 // Definitions by: Stanley Clark <me@stanrogo.com>
+import {Options} from './interfaces';
```
Comment (stanrogo) [inline src/index.d.ts]: This and a few other lines are not consistent with the linting rules set out in pritterrc and eslintrc. That's on me that I didn't include proper contribution guidelines (and the pre-commit hooks didn't seem to work). For now, please just accept my suggestions and I'll look into this at some point.

Comment (stanrogo) [inline src/index.d.ts]:
```suggestion

import { Options } from './interfaces';
```

File: src/index.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@
  * @file index.ts
  * @description Utility to read lines from files, without having to load the entire file into memory
  */
+import {Options} from './interfaces';
```
Comment (stanrogo) [inline src/index.ts]:
```suggestion

import { Options } from './interfaces';
```

File: src/interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (stanrogo) [inline src/interfaces.ts]:
```suggestion
```

File: src/interfaces.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+export interface Options {
+	encoding?: string;
+}
```
Comment (stanrogo) [inline src/interfaces.ts]:
```suggestion
}

```

**Review by stanrogo — CHANGES_REQUESTED** [review]
Nice and simple, I just have some linting issues and then I can merge it. Sorry for the delay @tonila, I hardly ever look at my repos. I should probably see if I can set appropriate email notifications.

**Review by stanrogo — APPROVED** [review]

**Pull request merged**